File: LC/Medium/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans = 0
        seen = set()
        window_start = 0
        
        for window_end in range(len(s)):
            while s[window_end] in seen:
                seen.remove(s[window_start])
                window_start += 1
            
            seen.add(s[window_end])
            ans = max(ans, window_end - window_start + 1)
        
        return ans
        

File: LC/C++/Medium/3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int ans = 0;
        int window_start = 0;
        unordered_set<char> seen = {};

        for (int i = 0; i < s.length(); i++) {
            while (seen.find(s[i]) != seen.end()) {
                seen.erase(s[window_start]);
                window_start++;
            }
            ans = max(ans, i - window_start + 1);
            seen.insert(s[i]);
        }

        return ans;
    }
};